### ISSUE TITLE
call doesn't work in certain cases

### ISSUE THREAD

**Issue by mhagmajer** [xFAANG/askql#46]
Probably `call` in AskScript has different semantics than in `AskCode` due to dereferencing of identifiers. This causes different issues:

**Example 1**
```
ask {
  call(fun { 5 })
}
```
returns the `AskCode` but `5` is expected

**Example 2**
```
ask {
  let f = fun { 5 }
  f
}
```
returns `5` but `AskCode` for `f` is expected

**Example 3**
```
ask {
  f
}
```
returns `null` but the resource for `f` is expected (currently shorthand for `fragment` which is used in complex clauses such as if or while)

**Comment by czerwinskilukasz1** [comment]
@mhagmajer, what's the status of this one? 
If it's not very advanced, I could take it because it's blocking my fix for lambda (#43 ).

**Comment by mhagmajer** [comment]
@czerwinskilukasz1 I already havr a draft PR as you can see that I'm going to finish today. There is a lot of other issues including your work on the terminal that you can do
